Lesson21/Fırat/app.js

```diff
@@ -1,51 +1,109 @@
-  // Select previous and next buttons
-  const previousButton = document.getElementById("prev-btn");
-  const nextButton =document.getElementById("next-btn");
-  // Select indicators
-  const indicators = document.getElementsByClassName('indicator');
 
+// DİFFERENT SOLVİNG OF THE PROBLEM
+// window.addEventListener("load", () => {
+//   // Select all images
+//   const images = document.querySelectorAll("img");
+//   // Select previous and next buttons
+//   const previousButton = document.getElementById("prev-btn");
+//   const nextButton = document.getElementById("next-btn");
+//   // Select indicators
+//   const indicators = document.getElementsByClassName("indicator");
+
+//   let currentIndex = 0;
+
+//   nextButton.addEventListener("click", () => {
+//     images[currentIndex].classList.remove("active");
+//     indicators[currentIndex].classList.remove("active");
+
+//     currentIndex++;
+//     if (currentIndex >= images.length) {
+//       currentIndex = 0;
+//     }
+
+//     images[currentIndex].classList.add("active");
+//     indicators[currentIndex].classList.add("active");
+//   });
+
+//   previousButton.addEventListener("click", () => {
+//     images[currentIndex].classList.remove("active");
+//     indicators[currentIndex].classList.remove("active");
+
+//     currentIndex--;
+//     if (currentIndex < 0) {
+//       currentIndex = images.length - 1;
+//     }
+
+//     images[currentIndex].classList.add("active");
+//     indicators[currentIndex].classList.add("active");
+//   });
+// });
+
+window.addEventListener("load", () => {
+  const images = document.querySelectorAll("img");
+  const previousButton = document.getElementById("prev-btn");
+  const nextButton = document.getElementById("next-btn");
+  const indicators = document.getElementsByClassName("indicator");
 
-  // Create application state to keep track of the current index
   let currentIndex = 0;
 
-  // Show images function
   function showImage(index) {
-    // Remove active class from all images
-    images.forEach(
-      // For every element in images array call a function that removes "active" class
-    )
+    images.forEach(function (image) {
+      image.classList.remove("active");
+    });
+    for (let i = 0; i < indicators.length; i++) {
+      indicators[i].classList.remove("active");
+    }
 
-    // Remove active class from all indicators
-    // Use loop to loop through all element of array
-
-    // Add active class to the current image and indicator
-    // images[]
+    if (index >= 0 && index < images.length) {
+      images[index].classList.add("active");
+      indicators[index].classList.add("active");
+      currentIndex = index;
+    }
   }
 
-  // handle next click
   function showNextImage() {
-    // increase currentIndex. What happens if the current index is > than number of images that we have?
-    currentIndex++;
-    showImage(currentIndex);
+    let newIndex = currentIndex + 1;
```

Comment:
In this function, you first assign a new value to newIndex, then check if it meets a condition, and override it.
It would make your code a bit cleaner if you check the condition first, then set the value of `newIndex` to whatever you'd need it to be.